tests/functional/test_custom_task_templates.py

```diff
@@ -206,3 +219,83 @@ def test_custom_task_templates(
             test_passed = True
 
         assert test_passed, "Job result is the object: " + str(res)
+
+    @pytest.mark.parametrize(
+        "template_type, model_template, proj, env, target_type, expected_msgs",
+        [
+            (
+                "fixture",
+                "validate_transform_fail_input_schema_validation",
+                "project_binary_diabetes",
+                "sklearn_drop_in_env",
+                "transform",
+                [
+                    "schema validation failed for input",
+                    "Datatypes incorrect. Data has types: NUM, but expected types to exactly match: CAT",
+                ],
+            ),
+            (
+                "fixture",
+                "validate_transform_fail_output_schema_validation",
+                "project_binary_diabetes",
+                "sklearn_drop_in_env",
+                "transform",
+                [
+                    "Schema validation found mismatch between dataset and the supplied schema",
+                    "Datatypes incorrect. Data has types: NUM, but expected types to exactly match: CAT",
+                ],
+            ),
+        ],
+    )
+    def test_custom_task_templates_fail__schema_validation(
+        self, request, template_type, model_template, proj, env, target_type, expected_msgs
+    ):
+        env_id, env_version_id = request.getfixturevalue(env)
+        proj_id = request.getfixturevalue(proj)
+        folder_base_path = self.get_template_base_path(template_type)
+
+        target_type = dr.enums.CUSTOM_TASK_TARGET_TYPE.TRANSFORM
+
+        custom_task = dr.CustomTask.create(name="transform", target_type=target_type)
+        custom_task_version = dr.CustomTaskVersion.create_clean(
+            custom_task_id=str(custom_task.id),
+            base_environment_id=env_id,
+            folder_path=os.path.join(folder_base_path, model_template),
+        )
+
+        # Only use numeric input features
+        w = Workshop()
+        bp = w.CustomTask(custom_task_version.custom_task_id, version=str(custom_task_version.id))(
+            w.TaskInputs.NUM
+        )
+        bp = w.Tasks.LR1()(bp)
+
+        user_blueprint = w.BlueprintGraph(bp).save()
```

Comment:
Also going to add an assert on blueprint validation - we should be able to see the input/output types at this stage

Reply:
BP workshop only supports jupyter notebook validation atm